### DISCUSSION TITLE
Is Output shape available in MyNewContribOp::Compute()?

### DISCUSSION THREAD

**Discussion by Dan5002** [microsoft/onnxruntime#5013] [Other Q&A]
I see that Input shapes are available in the context:
Status MyNewContribOp::Compute(OpKernelContext* context) const {
  auto X = context->Input<Tensor>(0);
  auto& dims = X->Shape();

But I would like that to be true for Outputs as well.

I don't want to hard code it:
  TensorShape output_shape_nms({1, 15130, 4});
  Tensor* output_to_nms = context->Output(1, output_shape_nms);

The shape is available when I graph my model, as my outputs are inputs to other nodes.  See screenshot of my model displayed with Netron.

Is there a way to parse the graph and find the expected shape of my output?  I don't see Output shape in the context.

![Node_output_shapes](https://user-images.githubusercontent.com/68922838/91909980-0bac9a80-ec63-11ea-9534-4cb56a9e47a7.jpg)

**Comment by Dan5002** [comment]
See comment in bold below.  This is just a restatement of the question in the prior post:

Is there a way to get the 'given tensor shape' from the graph, within the Compute() method (using context, I suppose)?

From include/onnxruntime/core/framework/op_kernel.h:
  
// Fetch output (non-tensor) with specified index.
  template <typename T>
  T* Output(int index) {
    if (index < 0 || index >= OutputCount())
      return nullptr;

    OrtValue* p_ml_value = GetOrCreateOutputMLValue(index);
    return p_ml_value ? p_ml_value->GetMutable<T>() : nullptr;
  }

  // In the case that memory allocation has not been done for an output tensor,
  // **The memory allocation will be done on-the-fly with given tensor shape.**
  // Return nullptr if the output is an unused optional output.
  Tensor* Output(int index, const TensorShape& shape);

**Comment by pranavsharma** [comment]
Currently there's no way to fetch the output shape from the graph using OpKernelContext.
